source/javascripts/components/ProductTour/Highlighter.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { Portal } from "./Portal";
+
+import "./Highlighter.scss";
+
+interface HighlighterProps {
+	clipPath: string;
+	rect: DOMRect;
+	children: JSX.Element;
+	isOpen: boolean;
+	renderSelection?: () => JSX.Element;
+}
+
+export const useHighlighter = (selectedId: string) => {
+	const ref = useRef<HTMLElement | null>(null);
+	const animationFrameRef = useRef<number | undefined>();
+	const [highlightedRect, setHighlightedRect] = useState<DOMRect | null>(null);
+
+	const highlightElement = useCallback(id => {
+		const element = document.getElementById(id);
+		if (element) {
+			ref.current = element;
+			const rect = element.getBoundingClientRect();
+			setHighlightedRect(rect);
+		}
+	}, []);
+
+	useEffect(() => {
+		if (selectedId) {
+			highlightElement(selectedId);
+		}
+	}, [selectedId, highlightElement]);
+
+	useEffect(() => {
+		const onResize = () => {
+			if (ref.current) {
+				if (animationFrameRef.current) {
+					cancelAnimationFrame(animationFrameRef.current);
+				}
+
+				animationFrameRef.current = requestAnimationFrame(() => {
```

Comment:
Why is `requestAnimationFrame` used here?
`setHighlightedRect` will be used only in the next render cycle of react, so does it add anything plus compared to using it without `requestAnimationFrame`?

Reply:
Completely agree, I was investigating some lagging/frame drops and I think I've left it in there.